hazelcast-jet-core/src/main/java/com/hazelcast/jet/config/JobConfig.java

```diff
@@ -418,6 +558,12 @@ public JobConfig addResource(@Nonnull String path, @Nonnull String id) {
     }
 
     private JobConfig add(URL url, String id, ResourceType resourceType) {
+        boolean duplicate = resourceConfigs.stream()
+                                           .map(ResourceConfig::getId)
+                                           .anyMatch(existingId -> existingId.equals(id));
+        if (duplicate) {
+            throw new IllegalArgumentException("Resource with id:" + id + " already exists!");
```

Comment:
can we remove the `!` marks from exceptions? There's several in many places in the PR